src/main/java/com/github/cameltooling/lsp/internal/parser/CamelKYamlDSLParser.java

```diff
@@ -0,0 +1,112 @@
+import org.yaml.snakeyaml.Yaml;
+
+import java.util.List;
+import java.util.Map;
+
+import com.github.cameltooling.lsp.internal.instancemodel.CamelURIInstance;
+import com.github.cameltooling.lsp.internal.instancemodel.YamlDSLModelHelper;
+
+/**
+ * @author Lars Heinemann
+ */
+public class CamelKYamlDSLParser extends ParserFileHelper {
+	
+	public static final String URI_KEY = "uri";
+	public static final String REST_KEY = "rest";
+	public static final String FROM_KEY = "from";
+	public static final String TO_KEY = "to";
+
+	@Override
+	public String getCamelComponentUri(String line, int characterPosition) {
+		String camelComponentURI = null;
+		Map data = parseYaml(line);
+		if (data.containsKey(URI_KEY) && URI_KEY.length() < characterPosition) {
+			camelComponentURI = data.get(URI_KEY).toString();
+		} else if(data.containsKey(TO_KEY) && data.get(TO_KEY).toString().trim().length()>0) {
+			camelComponentURI = data.get(TO_KEY).toString();
+		}
+		return camelComponentURI;
+	}
+
+	@Override
+	public CamelURIInstance createCamelURIInstance(TextDocumentItem textDocumentItem, Position position,
+			String camelComponentUri) {
+		CamelURIInstance uriInstance = new CamelURIInstance(camelComponentUri, new YamlDSLModelHelper(getCorrespondingType(textDocumentItem, position.getLine())), textDocumentItem);
+		int start = getStartCharacterInDocumentOnLinePosition(textDocumentItem, position);
+		uriInstance.setStartPositionInDocument(new Position(position.getLine(), start));
+		uriInstance.setEndPositionInDocument(new Position(position.getLine(), start+camelComponentUri.length()));
+		return uriInstance;
+	}
+
+	private int getStartCharacterInDocumentOnLinePosition(TextDocumentItem textDocumentItem, Position position) {
+		String line = parserFileHelperUtil.getLine(textDocumentItem, position.getLine());
+		String uri = extractUriFromYamlData(line);
+		return line.indexOf(uri);
+	}
+
+	@Override
+	public int getPositionInCamelURI(TextDocumentItem textDocumentItem, Position position) {
+		String line = parserFileHelperUtil.getLine(textDocumentItem, position.getLine());
+		return position.getCharacter() - line.indexOf('"')-1;
+	}
+
+	public String getCorrespondingType(TextDocumentItem textDocumentItem, int lineNumber) {
+		for (int lineNo = lineNumber; lineNo >=0; lineNo--) {
+			String tempLine = parserFileHelperUtil.getLine(textDocumentItem, lineNo);
+			Map data = parseYaml(tempLine);
+			if (data.containsKey(TO_KEY)) {
+				return "to";
+			} else if (data.containsKey(FROM_KEY)) {
+				return "from";
+			} else if (data.containsKey(REST_KEY)) {
+				return null;
+			}
+		}
+		return null;
+	}
+
+	private Map parseYaml(String line) {
+		Yaml yaml = new Yaml();
+		Object obj = yaml.load(line);
+		return extractMapFromYaml(obj);
+	}
+
+	private String extractUriFromYamlData(String line) {
+		Yaml yaml = new Yaml();
+		Object obj = yaml.load(line);
+		Map m = extractMapFromYaml(obj);
+		return (String)m.values().toArray()[0];
```

Comment:
might need to check that there is not an empty map

Reply:
done